hugolib/page_collections.go

```diff
@@ -215,6 +216,10 @@ func (c *PageCollections) getPageNew(context *Page, ref string) (*Page, error) {
 	if !strings.HasPrefix(ref, "/") {
 		// Many people will have "post/foo.md" in their content files.
 		if p, err := c.getFromCache("/" + ref); err == nil && p != nil {
+			if context != nil {
+				// TODO(bep) remove this case and the message below when the storm has passed
+				helpers.DistinctFeedbackLog.Printf(`WARNING: make non-relative ref/relref page reference(s) in page %q absolute, e.g. {{< ref "/blog/my-post.md" >}}`, context.absoluteSourceRef())
```

Comment:
Hard to write a succinct warning because there are four cases this happens, three of which need very different warnings, and the fourth needing no warning at all!:

1.  The path is intended to be absolute. Tell user to use an absolute path to remove ambiguity and the warning.
2. The path is intended to be relative, but no page was found at that path relative to the page. So instead we are returning the wrong page. Warn user of this.
3. The path is an ambiguous "shorthand" path such as "intro.md", and there happens to be one "intro.md" at the root, so we are returning that one without an ambiguity error (the prepended "/" avoids the ambiguity check)
4. The path is an unambiguous "shorthand" path such as "about.md", and "about.md" happens to be in the root, so we are actually returning the right page. 

I'll think about it. We can also just put something generic in and update it later.  We could also isolate #4 so that it doesn't get the warning.

Reply:
Yea, I thought about adding the ref to the error message, but it gets "floody and wordy very fast". This has to be documented somewhere else, too.